packages/app/src/debug/DebugPageDetails.vue

```diff
@@ -0,0 +1,67 @@
+<template>
+  <div
+    class="flex flex-col w-full px-24px justify-center items-center align-middle"
+    :class="{'flex-grow': ['PASSED', 'OVERLIMIT'].includes(status)}"
+  >
+    <DebugCancelledAlert
+      v-if="status === 'CANCELLED'"
+      :total-specs="totalSpecs"
+      :total-skipped-specs="totalSkippedSpecs"
+      :canceled-at="canceledAt"
+      :canceled-by-full-name="canceledByFullName"
+      :canceled-by-email="canceledByEmail"
+    />
+    <DebugPassed v-else-if="status === 'PASSED'" />
+    <DebugErrored
+      v-else-if="status === 'ERRORED'"
+      :errors="errors"
+      :total-specs="totalSpecs"
+      :total-skipped-specs="totalSkippedSpecs"
+    />
+    <DebugNoTests v-else-if="status === 'NOTESTS'" />
+    <DebugTimedout
+      v-else-if="status === 'TIMEDOUT'"
+      :total-specs="totalSpecs"
+      :total-skipped-specs="totalSkippedSpecs"
+      :ci="ci"
+    />
+    <DebugOverLimit
+      v-else-if="status === 'OVERLIMIT' || isHiddenByUsageLimits"
+      :over-limit-action-type="overLimitActionType"
+      :over-limit-action-url="overLimitActionUrl"
+    />
+  </div>
+</template>
+<script lang="ts" setup>
+import { computed } from '@vue/reactivity'
+import type { CloudRunStatus, OverLimitActionTypeEnum } from '../generated/graphql'
+import DebugCancelledAlert from './DebugCancelledAlert.vue'
+import DebugPassed from './DebugPassed.vue'
+import DebugErrored from './DebugErrored.vue'
+import DebugNoTests from './DebugNoTests.vue'
+import DebugTimedout from './DebugTimedout.vue'
+import DebugOverLimit from './DebugOverLimit.vue'
+import type { CloudCiBuildInfo } from '@packages/data-context/src/gen/graphcache-config.gen'
+
+const props = defineProps<{
+  status: CloudRunStatus
+  canceledAt: string | null
+  canceledByFullName?: string | null
+  canceledByEmail?: string | null
+  isHiddenByUsageLimits: boolean
+  overLimitActionType: OverLimitActionTypeEnum
+  overLimitActionUrl: string
+  specs: readonly any[]
```

Comment:
I couldn't get the types to work for this. I wanted it to be of type `CloudSpecRun[]` but I wasn't sure where to import this type from

Reply:
Should we just make this a GraphQL component that has a fragment? It seems we are passing quite a bit of data to it, anyway.

Reply:
I think it would be

```ts
import type { CloudRunStatus, DebugPageFragment, OverLimitActionTypeEnum } from '../generated/graphql'

specs: DebugPageFragment[]
```

Tried it, looks like `CloudRunStatus` is 

```ts
export type CloudRunStatus =
  | 'CANCELLED'
  | 'ERRORED'
  | 'FAILED'
  | 'NOTESTS'
  | 'OVERLIMIT'
  | 'PASSED'
  | 'RUNNING'
  | 'TIMEDOUT';
```

Does not include `UNCLAIMED` - maybe my Cloud repo is out of date, or is this type incorrect?

Reply:
Maybe we make this have a GraphQL fragment and then get the types that way.

Reply:
So there are two different status types:

`CloudRunStatus` and `CloudSpecStatus`

`CloudRunStatus` is the status of the overall run, while `CloudSpecStatus` is the status of an individual spec that was in the run. They overlap quite a bit so it can be confusing. `spec.status` is of type `CloudSpecStatus`, which **does** have `UNCLAIMED`.